Add R16 release notes (#210)
pxblue/doc-it: src/docs/release-notes/R16/R16.mdx

### Documentation Site

Added search capability to doc site

### New Components

We have added new components to our [Angular](https://pxblue-components.github.io/angular) and [React Native](https://pxblue-components.github.io/react-native) component libraries:

> -   For Angular, we have added components for [Drawer](https://pxblue-components.github.io/angular-dev/?path=/info/components-drawer--readme), [ListItemTag](https://pxblue-components.github.io/angular-dev/?path=/info/components-list-item-tag--readme), [Spacer](https://pxblue-components.github.io/angular-dev/?path=/info/components-spacer--readme), and

Reviewer: ```suggestion
-   For Angular, we now have components for [Drawer](https://pxblue-components.github.io/angular-dev/?path=/info/components-drawer--readme), [ListItemTag](https://pxblue-components.github.io/angular-dev/?path=/info/components-list-item-tag--readme), [Spacer](https://pxblue-components.github.io/angular-dev/?path=/info/components-spacer--readme), and
```

> -   For Angular, we have added components for [Drawer](https://pxblue-components.github.io/angular-dev/?path=/info/components-drawer--readme), [ListItemTag](https://pxblue-components.github.io/angular-dev/?path=/info/components-list-item-tag--readme), [Spacer](https://pxblue-components.github.io/angular-dev/?path=/info/components-spacer--readme), and

Reviewer: In our documentation we're referring to these as `<pxb-drawer>` etc. Do we want to do that here, or stick with the TitleCase?

Author: I think for our release notes we should stick to TitleCase since that's what our earlier releases have been using and it looks fine to me.

>     [Scorecard](https://pxblue-components.github.io/angular-dev/?path=/info/components-score-card--readme), and [InfoListItem](https://pxblue-components.github.io/angular-dev/?path=/info/components-info-list-item--readme).

Reviewer 2: These are pointing to the storybook dev site.

> -   For React Native, we now have a [Drawer](https://pxblue-components.github.io/react-native/?path=/info/components-documentation--drawer) component.

Reviewer 2: These are pointing to the storybook dev site.

Author: I put these links in to put to dev site because they'll fail our CI build until our published storybook site has the new components.  An alternate solution would be to remove the links until the storybook is released.

Reviewer: I'll happily merge the PR in with these broken links

Author: Ok I've swapped out all dev links for the master version.

> -   For React Native, we have added a [Drawer](https://pxblue-components.github.io/react-native/?path=/info/components-documentation--drawer) component.

Reviewer: ```suggestion
-   For React Native, we now have a [Drawer](https://pxblue-components.github.io/react-native/?path=/info/components-documentation--drawer) component.
```

> ### React Native Reusable Workflow

Reviewer: ```suggestion
### Reusable Workflows
```

> Innersourced a reusable [Authentication Workflow](https://github.com/pxblue/react-native-workflows) for React Native apps

Reviewer: ```suggestion
We have finished innersourcing a reusable [Authentication Workflow](https://github.com/pxblue/react-native-workflows) for React Native apps, allowing teams to use a common implementation across different projects.
```

Author: I've used this edited version but have taken out the React Native Auth workflow link since the repo is private and this'll fail our pipeline.

> We have finished innersourcing a reusable Authentication Workflow for React Native apps, allowing teams to use a common implementation across different projects.

Reviewer 2: Link to the NPM package page.

Author: Ah good idea, I'll add a link to the NPM package.

### Design Resources

> Published our stickersheet on Figma for public viewing

Reviewer: ```suggestion
Published our stickersheet to the Figma community
```

> ### Miscellaneous

Author: Is this neccessary?

Reviewer: I don't think it hurts anything to have it in there. Most of the time we are fixing bugs and stuff, so it might be nice for people to see that we are doing maintenance on existing materials. Some other libraries like to call out specific bugs that are fixed in each release...that might be a little overkill for us.

Author: Sounds good to me!  If people are ever asking about bug fix specifics I think this is something that we can integrate into github via tags instead of through our release notes.

    [Scorecard](https://pxblue-components.github.io/angular-dev/?path=/info/components-score-card--readme), and [InfoListItem](https://pxblue-components.github.io/angular-dev/?path=/info/components-info-list-item--readme).
-   For React Native, we now have a [Drawer](https://pxblue-components.github.io/react-native/?path=/info/components-documentation--drawer) component.

### Reusable Workflows

We have finished innersourcing a reusable [Authentication Workflow](https://www.npmjs.com/package/@pxblue/react-native-auth-workflow) for React Native apps, allowing teams to use a common implementation across different projects.

### Design Resources

Published our stickersheet to the Figma community.

### Design Patterns

We have migrated a majority of our React design patterns into a [single repository](https://codesandbox.io/s/github/pxblue/react-design-patterns). Angular and React Native design patterns will be migrated into their own repos in R17.

### PX Blue CLI

> Introduced the [PX Blue CLI](https://www.npmjs.com/package/@pxblue/cli), a tool used to generate new Angular, React, or React Native projects loaded with PX Blue components and themes.

Reviewer: It also does Ionic

Author: Aha, updated.
